CT-1764 Ensure reports reflect responding teams changed after closure

stephenrichards opened the pull request:
A new feature has been enabled to allow BMT to change the responding team after
case closure.  This POR ensures that the new responding team is properly reflected
in all the reports.

In fact, this already happens: the Case::Base#responding_team method which is what the
reports use to determine which BU to count the case against will already return the
new team.  So this PR simply adds a specific spec to test that method in the case where
a responding team has been changed after closure.

misaka commented on lib/cts/demo_setup.rb:
```diff
@@ -0,0 +1,137 @@
+    def initialize(number)
+      @number = number
+      @cases = Case.order(id: :desc).limit(number).to_a
+      @managing_team = Team.where(name: 'DACU').first
```
Just for reference, `CTS::dacu_team` is a short-cut that standardises finding the DACU team.

stephenrichards committed:
CT-1764 Ensure reports reflect responding teams changed after closure

A new feature has been enabled to allow BMT to change the responding team after
case closure.  This POR ensures that the new responding team is properly reflected
in all the reports.

In fact, this already happens: the Case::Base#responding_team method which is what the
reports use to determine which BU to count the case against will already return the
new team.  So this PR simply adds a specific spec to test that method in the case where
a responding team has been changed after closure.

beccapearce committed:
Merge branch 'master' into ct-1764-reports-reflect-responder-assignmetns-after-closure

beccapearce committed:
Merge branch 'master' into ct-1764-reports-reflect-responder-assignmetns-after-closure
